### ISSUE TITLE
Arcus request middleware should log requests without specific route-parameters

### ISSUE THREAD

**Issue by fgheysels** [arcus-azure/arcus.webapi#407]
I'm testing Arcus.WebApi.Logging 1.7.0-preview-3, more specifically the request logging middleware and I'm writing to application insights.

When I perform some requests to my API, those requests appear in the 'Performance' overview in Application Insights like this:

![image](https://user-images.githubusercontent.com/3605786/204032671-342ec290-b981-4af8-9e65-15db6f1a8874.png)

This is not good, as this means that the 'Performance' overview will contain an entry for each request where the route-parameters are different.

When you would use the default Microsoft middleware for requestlogging, the requests are logged like this:

![image](https://user-images.githubusercontent.com/3605786/204032814-ea62ba97-a545-464d-a9ca-ebf51856048a.png)

This is better, as all requests to the same operation are grouped together, regardless the route-parameters.

**Comment by stijnmoreels** [comment]
Ok, that is more a feature that we allow via the HTTP request options to override the operation name (that is currently now generated based on the exact route). Thx.

**Comment by fgheysels** [comment]
I think it is mandatory to have this in the 1.7 release, as request logging in APIM is otherwise really unusable.

**Comment by stijnmoreels** [comment]
> I think it is mandatory to have this in the 1.7 release, as request logging in APIM is otherwise really unusable.

Ok, could you take this up? I can then look for the correlation failure (I think I found it why it happened).

**Comment by fgheysels** [comment]
I can see if I can spend some time on this later this week.

> ou can use the httpContext.GetEndpoint() as RouteEndpoint to retrieve the .RoutePattern.RawText, but maybe we should have an option to control whether to use this route pattern or not (default: true) as there are scenarios where you want to full HTTP request path (including query parameter values) as operation name). 
To be honest, I don't see such scenario's ?  All requests to that endpoint should be grouped together, regardless the route-parameters and the route-parameters should just be available as properties of each request.

**Comment by stijnmoreels** [comment]
Otherwise I can see if I can pick it up between things.

**Comment by fgheysels** [comment]
Looks like this needs to be done in Arcus.Observability, and not in Arcus.WebApi.

**Comment by stijnmoreels** [comment]
> Looks like this needs to be done in Arcus.Observability, and not in Arcus.WebApi.

No, I don't think so. We need to retrieve the route pattern from the HTTP endpoint and send that as `operationName` to the `LogRequest` call in the HTTP Web API middleware. Observability is not to be changed, right?

**Comment by fgheysels** [comment]
These things are written in Observability ?
https://github.com/arcus-azure/arcus.observability/blob/1cef318d2afb01543d65c655e0e32462b51708e7/src/Arcus.Observability.Telemetry.AspNetCore/Extensions/ILoggerRequestExtensions.cs#L393

**Comment by fgheysels** [comment]
You might be right.  I'll continue looking at it.
However, I guess we should also build up a string that looks like '{HttpMethod} {EndpointName}/{RouteParamNames}

**Comment by stijnmoreels** [comment]
This is already kind of done when no operation name is present: https://github.com/arcus-azure/arcus.observability/blob/1cef318d2afb01543d65c655e0e32462b51708e7/src/Arcus.Observability.Telemetry.Core/Logging/RequestLogEntry.cs#L163
